### DISCUSSION TITLE
How can sorting be implemented on Microsoft Search Data source?

### DISCUSSION THREAD

**Discussion by mw-new-test** [microsoft-search/pnp-modern-search#3260] [Q&A] [answered]
For sorting Graph connector results (externalItem), sort settings are not available. 
Sorting only works for SharePoint data source.

Has anyone implemented sorting for externalItem? - Not just **Default sort** but **User sort** option as well.

Also, is there any Graph API limitation to sort externalItems?

**Comment by wobba** [comment, marked as answer]
This is a known limitation https://learn.microsoft.com/en-us/graph/api/resources/search-api-overview?view=graph-rest-1.0#sort-search-results

**Reply by gretchunkim** [reply]
@wobba , just to clarify, is the limitation not to be overcome by using handlebars-helpers ([sort](https://github.com/helpers/handlebars-helpers#sort), [sortBy](https://github.com/helpers/handlebars-helpers#sortBy))?

**Reply by gretchunkim** [reply]
@mw-new-test What is your data source (using which  MS built/released connector? I'm wondering if there is a workaround if a  custom connector is built and used instead MS built connectors (Azure Devops, Jira, ADLS GEn2, Oracle SQL database, ServiceNow, File share, CSV, MS SQL, MediaWiki, Salesforce, Confluence, Azure SQL)?  I'm using MS SQL, and it doesn't allow sort of result even it is sorted stored procedure method. 

According to this Graph API documentation, sorting is supported in external items.  
https://learn.microsoft.com/en-us/graph/search-concept-sort#known-limitations 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/8564035/c45ff4ab-a767-4bc2-bd5a-8aee3c1c4ce1)

**Reply by mw-new-test** [reply]
Thanks for sharing @gretchunkim.
We are using Confluence cloud MS built-in connector.
I found this fix - https://github.com/microsoft-search/pnp-modern-search/pull/2150 but it only works for default sort to sort by Last Modifed date.
I believe, user enabled sorting is an enhancement yet to be developed for externalItem.
Also, sorting on interleaved results is a limitation as per MS documentation 
https://learn.microsoft.com/en-us/graph/search-concept-interleaving

**Reply by wobba** [reply]
@gretchunkim my bad, I misread what I linked. Not sure how I managed to. So yes, sorting should work on properties which are set as refinable for external items.

**Reply by gretchunkim** [reply]
@wobba it’s the documentation that’s confusing… one place it does say it’s not supported then another place it does. It’s hard to tell until one start dissect each word and try. Btw, thank you for this awesome webpart and support from this project owners and contributors!

**Reply by wobba** [reply]
Indeed..I'll see if I can get it sorted 🙂

**Reply by gretchunkim** [reply]
A new issue created to resolve conflicting Microsoft documentation information about know limitation of external items sorting feature. 
https://github.com/microsoftgraph/microsoft-graph-docs-contrib/issues/9065
